### ISSUE TITLE
CA1802 false positive on nint

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn-analyzers#3662]
#### Analyzer package

Unknown

#### Package Version

dotnet/roslyn@fa4ffc8

#### Diagnostic ID

Example: [CA1802](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1802)

#### Repro steps

```
internal static readonly IntPtr NullPtr = (nint)0;
```

#### Expected behavior

CA1802 is not reported.

#### Actual behavior

CA1802 is reported, and the code fix results in a compiler error.

**Comment by Evangelink** [comment]
@sharwell I am not sure how to add a unit test for this.

The following test does not report any issue:

```csharp
[Fact]
public async Task CA1802_CSharp_nint_NoDiagnostic()
{
	await VerifyCS.VerifyAnalyzerAsync(@"
using System;

namespace System
{
	public struct nint
	{
		public static implicit operator IntPtr(nint n) => new IntPtr(0);
		public static explicit operator nint(int i) => new nint();
	}
}

public class Class1
{
	internal static readonly IntPtr NullPtr = (nint)0;
}");
}
```

It's not considered as a constant value.

**Comment by Evangelink** [comment]
I was giving a fresh look at this ticket and now I can easily reproduce using:

```csharp
[Fact]
public async Task CA1802_CSharp_nint_NoDiagnostic()
{
	await new VerifyCS.Test
	{
		LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
		ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
		TestCode = @"
using System;

public class Class1
{
	internal static readonly IntPtr NullPtr = (nint)0;
}",
	}.RunAsync();
}
```

but looking at the implementation I am actually wondering if the fix should be done around the `nint` or whether we should simply exclude some types that can't be constant (in this case `IntPtr`). WDYT?
